### DISCUSSION TITLE
How to help?

### DISCUSSION THREAD

**Discussion by glthm** [dmossakowski/skala3ma#37] [Q&A]
Hi, I want to help on this project, I'm not a very experienced Github user but I work with Python everyday and I have the bases of HTML

What can I do?

**Comment by dmossakowski** [comment]
Hi, unfortunately you will need to know git and GitHub at least a little bit to work on the project. I encourage you to learn and I can help answer questions.

If you know Python the perhaps you could help on making skala_api.py more object oriented. I started with creation of a simple class Activity.py and also looked.at pydantic. The goal would be to reduce the code manipulating dictionaries and lists in different places. Take a look at skala_api.py and skala_db.py to see what I mean. Ideally we should be manipulating Activity, Gym, Route and.other classes

**Reply by glthm** [reply]
I know the bases of git and github but that's pretty much it ; I'll look into the files you gave soon, I have some experience with typing and OOP

**Comment by dmossakowski** [comment]
Sounds good. Base knowledge is all you need.

Thanks in advance!

**Comment by GrizzlyFr** [comment]
Hi @dmossakowski!

Do you think you can prepare a mockup "competitions.sqlite" file with redacted personnal info? I set my instance of Skala3ma and I think I can intervene on the "closed competition result pdf" topic. But it would be easier with an example db.

**Comment by dmossakowski** [comment]
Yes this is a good idea. I will try to get something done. In the meantime, you should be able to create a gym and then a competition and register some users to it, give them scores and so on.  If you don't see the button to create a gym, modify your user in sqlite to add:   

`"general": ["edit_competition", "create_competition", "update_routes", "create_gym"],`

**Reply by GrizzlyFr** [reply]
I tested all the functions. I properly have these rights in sqlite.
